include/ofi_bitmask.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2019-2019 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _OFI_BITMASK_H_
+#define _OFI_BITMASK_H_
+
+#include <rdma/fi_errno.h>
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <string.h>
+
+struct bitmask {
+	ssize_t size;
+	uint8_t *bytes;
+};
+
+static inline int ofi_bitmask_init(struct bitmask *mask, ssize_t size)
+{
+	mask->bytes = calloc(size, 1);
+	if (!mask->bytes)
+		return -FI_ENOMEM;
+
+	mask->size = size;
+
+	return FI_SUCCESS;
+};
+
+static inline void ofi_bitmask_free(struct bitmask *mask)
+{
+	if(mask->bytes)
+		free(mask->bytes);
+
+	mask->bytes = NULL;
+};
+
+static inline void ofi_bitmask_unset(struct bitmask *mask, ssize_t idx)
+{
+	assert(idx <= (mask->size * 8));
+	mask->bytes[idx / 8] &= ~(0x01 << (idx % 8));
+};
+
+static inline void ofi_bitmask_set(struct bitmask *mask, ssize_t idx)
+{
+	assert(idx <= (mask->size * 8));
+	mask->bytes[idx / 8] |= (0x01 << (idx % 8));
+};
+
+static inline void ofi_bitmask_set_all(struct bitmask *mask)
+{
+	int cur = mask->size;
+	while (cur) {
+		mask->bytes[--cur] = 0xff;
+	}
+};
+
+static inline ssize_t ofi_bitmask_get_lsbset(struct bitmask mask)
```

Comment:
Either return size_t, or a negative error if no bits are set.

Also, would it be more efficient to have a set/clear lsb call, rather than a get followed by set/clear?  Though this can be deferred until later.